Shinez Bid Adapter: update adapter to comply with latest Prebid version

Soterix opened the pull request:
## Type of change
- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
A refactored version of the Shinez bid adapter.

- test parameters for validating bids
```javascript
var adUnits = [{
        code: 'dfp-video-div',
        sizes: [
            [640, 480]
        ],
        mediaTypes: {
            video: {
                playerSize: [
                    [640, 480]
                ],
                context: 'instream'
            }
        },
        bids: [{
            bidder: 'shinez',
            params: {
                org: '56f91cd4d3e3660002000033', // Required
                floorPrice: 2.00, // Optional
                placementId: 'shinez-video-test', // Optional
                testMode: true // Optional
            }
        }]
    },
    {
        code: 'dfp-banner-div',
        sizes: [
            [640, 480]
        ],
        mediaTypes: {
            banner: {
                sizes: [
                    [640, 480]
                ]
            }
        },
        bids: [{
            bidder: 'shinez',
            params: {
                org: '56f91cd4d3e3660002000033', // Required
                floorPrice: 2.00, // Optional
                placementId: 'shinez-banner-test', // Optional
                testMode: true // Optional
            }
        }]
    }
];
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer: tech-team@shinez.io
- [x] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:
- A link to a PR on the docs repo at: https://github.com/prebid/prebid.github.io/pull/3808

## Other information

Soterix commented:
PR was created 7 days ago, could anybody take a look at it, please?

Soterix commented:
Guys, would anybody take a look at this PR, please?

Soterix commented:
Rebased on the last state of master, anything else I can do to speed up the review process?

gitsh84 commented:
Hey, any update re this PR ?

Soterix commented:
@ChrisHuie could you please update us on how we can get this approved?

ChrisHuie commented:
@Soterix @gitsh84 my apologies. Our integration tests didn't seem to run on this pr. Can you please give me edit access so I can try and get them to run? They automatically run on new prs but you are likely following your fork of Prebid in CircleCi instead of Prebid itself.

Soterix commented:
@ChrisHuie, thanks for update. Unfortunately, I am not able to allow editing for you because of https://github.com/orgs/github-community/discussions/5634. Instead, I have rebased changes on the last state of the upstream master and fixed failed tests. Is everything ok now?

ChrisHuie commented:
@Soterix so our CircleCI tests still didn't run on this pr. It has to do with your github account linking to your fork of the project in CircleCI vs the main project. You have to unfollow your fork in CircleCi I beleive. I can just resubmit this pr and the testing should pop up though if you are cool with that?

Soterix commented:
understand... @ChrisHuie, can you advise where it is possible to change CircleCi configuration (linking)? I have tried to log in to https://app.circleci.com/ with my GitHub account but I can't find anything related to Prebid. Maybe the issue is that fork is not under the personal account but the organization one? Do we need to allow integration between CircleCi and our organization account? 
Alternatively, I can try to do everything from scratch, I mean remove the repository fork, fork again, add an adapter and create a new PR. Do you think it will resolve the issue with CircleCi?

ChrisHuie commented:
> @ChrisHuie, you have mentioned that you can re-submit this PR, which should also be ok for us. Could you please do it?

Sure. Will submit and copy you both today. Will make 2 prs. One for master and one for Prebid 6 legacy branch.

patmmccann commented on modules/shinezBidAdapter.js:
```diff
@@ -0,0 +1,430 @@
+
+  if (bidderRequest && bidderRequest.refererInfo) {
+    generalParams.referrer = deepAccess(bidderRequest, 'refererInfo.referer');
+    generalParams.page_url = config.getConfig('pageUrl') || deepAccess(window, 'location.href');
```
this isn't valid behavior in 7

patmmccann replied:
just look for site.page

patmmccann commented on modules/shinezBidAdapter.js:
```diff
@@ -0,0 +1,430 @@
+  }
+
+  if (bidderRequest && bidderRequest.refererInfo) {
+    generalParams.referrer = deepAccess(bidderRequest, 'refererInfo.referer');
```
look for site.ref

Soterix replied:
Hey @patmmccann, as i understand `ortb2.site.page` and `ortb2.site.ref` should be for Prebid 7 - right?

Soterix commented:
@patmmccann, @ChrisHuie updated, and it looks like `ci/circleci` tests were run on this PR. Can it be merged?

patmmccann reviewed (commented):
Lgtm
